src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/HotSpotResolvedJavaMethodImpl.java

```diff
@@ -632,7 +632,7 @@ public LocalVariableTable getLocalVariableTable() {
 
         for (int i = 0; i < localVariableTableLength; i++) {
             final int startBci = UNSAFE.getChar(localVariableTableElement + config.localVariableTableElementStartBciOffset);
-            final int endBci = startBci + UNSAFE.getChar(localVariableTableElement + config.localVariableTableElementLengthOffset);
+            final int endBci = startBci + UNSAFE.getChar(localVariableTableElement + config.localVariableTableElementLengthOffset) - 1;
```

Comment:
Just a question: Can the length of a local variable be 0?

**If the code length is 0, the `endBci` here may be less than `startBci`.**

Reply:
I don't see anything in [JVMS 4.7.13](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.7.13) that says it cannot be 0. It basically means the LVT entry is useless (denotes a local that is never alive) but is otherwise harmless.
Maybe add this to the javadoc for `getEndBci()` to make the API user aware of this corner case:
```
     * If the value returned is less than {@link #getStartBCI}, this object denotes a local that is never live.
```

Reply:
The reason, which causes this problem, is that the `Local::endBci` includes itself instead of excluding it. But now, we can only fix the javadoc just as you suggested.

> If the value returned is less than {@link #getStartBCI}, this object denotes a local that is never live.

`a local variable` may be better to `a local` above.

Reply:
I had checked the specs on that and came to the same conclusion.
I also think the current state is fine in that regards in terms of code since it just means that there is no bci where this local would be valid when checking both start and end bci.
Adding a note about that to the javadoc is a good idea. I'll do that.